### PR TITLE
charts/meshconfig: make port and IP range validations stricter

### PR DESCRIPTION
**Pull request by shashankram**
**Description**:
Makes the port and IP range validations stricter.

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [X]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
`No`

### CODE REVIEW

File: charts/osm/crds/meshconfig.yaml [hunk 1]
```diff
@@ -97,13 +97,13 @@ spec:
...
                         type: string
-                        pattern: ((?:\d{1,3}.){3}\d{1,3})\/(\d{1,2})$
+                        pattern: ((?:\d{1,3}\.){3}\d{1,3})\/(\d{1,2})$
                     outboundPortExclusionList:
                       description: Global list of ports to exclude from outbound traffic interception by the sidecar proxy.
                       type: array
                       items:
                         type: integer
-                        pattern: ^[1-9]\d*$
+                        pattern: ^([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$ # 1 to 65535
```
Comment (nojnhuh) [inline charts/osm/crds/meshconfig.yaml:106]: Would `minimum`/`maximum` vs. `pattern` work here?
https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/#JSONSchemaProps

**Review by nojnhuh — APPROVED** [review]

File: charts/osm/crds/meshconfig.yaml (hunk 1 above)
Comment (shashankram) [reply charts/osm/crds/meshconfig.yaml:106]: It shows the types as `double`, so not certain it would produce the desired effect.

**Review by snehachhabria — APPROVED** [review]

File: charts/osm/crds/meshconfig.yaml (hunk 1 above)
Comment (shashankram) [reply charts/osm/crds/meshconfig.yaml:106]: Going to follow-up on this and address if possible in a subsequent PR.

Comment (shashankram) [reply charts/osm/crds/meshconfig.yaml:106]: https://github.com/openservicemesh/osm/pull/3337

**Pull request merged**